tlscanary/template/js/report_page.js

```diff
@@ -185,7 +174,7 @@ function makeMetaTab(meta)
   html += "</tr>";
   html += "</thead>";
 
-  for (var i=0;i<metaArray.length;i++)
+  for (var i = 0;i < metaArray.length;i++)
```

Comment:
You might actually want to check if one of the modern array iterators like `Array.forEach`, `for x in` or `for x of` appeal to you here. There are subtle differences in the way they work, but they should all work fine in this case. They all get rid of the unused iterator variable. Should work for a few other of your for loops as well.

Reply:
I will try it. My experience with `Array.forEach` is that - at least in previous ECMAScript implementations - order is not guaranteed. In our project, order is important. The docs say it goes in ascending order, so if that's true, then you are right, we should be using it.

Reply:
Oh, I thought order was only questionable in objects, but Array ordering is strict.

Reply:
OK, after trying to rewrite the `for` loops, I now remember why this isn't a good idea. They tend to enumerate unwanted properties. This is expected for a generic object, but for an indexed array, you get the indexed items AND other weird, undesirable junk that shows up due to things about JS that I don't understand. Regardless, I wasn't able to replace the loops for this reason, so I am going to leave this as-is.

Reply:
Sounds to me like the difference between `for in` and `for of`. `for of` should both conserve sequence and only iterate just the iterables. Have you tried?

Anyway, nothing to lose sleep over. Using the index is fine, too.

Reply:
(see [MDN on for..of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of))

Reply:
Good to know, and thank you for the additional info. I will keep this in mind.